File: admin/src/services/adminAuth.ts
import { z } from "zod";
import { request } from "@/services/apiClient";

const sessionSchema = z.object({
  access_token: z.string(),
  refresh_token: z.string(),
  access_expires_at: z.string(),
  refresh_expires_at: z.string(),
  display_name: z.string(),
  capability_label: z.string(),
});

const meSchema = z.object({
  display_name: z.string(),
  capability_label: z.string(),
  session_expires_at: z.string(),
  environment_kind: z.enum(["demo", "authorized", "unconfigured"]),
});

export interface AdminSessionSummary {
  displayName: string;
  capabilityLabel: string;
  environmentKind: "demo" | "authorized" | "unconfigured";
  accessToken: string;
  refreshToken: string;
  accessExpiresAt: string;
  refreshExpiresAt: string;
  sessionExpiresAt?: string;
}

export const FIXED_ADMIN_LOGIN = "心理健康中心工作人员";
export const FIXED_ADMIN_CAPABILITY = "超级管理员";

export async function loginAdmin(
  password: string,
): Promise<AdminSessionSummary> {
  const result = await request(
    "/api/v1/admin/auth/login",
    { method: "POST", body: { login_name: FIXED_ADMIN_LOGIN, password } },
    (value) => sessionSchema.parse(value),
  );
  const session = toSession(result.data);
  const me = await loadAdminMe(session.accessToken);
  return { ...session, ...me };
}

export async function loadAdminMe(
  accessToken: string,
): Promise<
  Pick<
    AdminSessionSummary,
    "displayName" | "capabilityLabel" | "environmentKind" | "sessionExpiresAt"
  >
> {
  const result = await request(
    "/api/v1/admin/me",
    { token: accessToken },
    (value) => meSchema.parse(value),
  );
  return {
    displayName: result.data.display_name,
    capabilityLabel: result.data.capability_label,
    environmentKind: result.data.environment_kind,
    sessionExpiresAt: result.data.session_expires_at,
  };
}

export async function logoutAdmin(accessToken: string): Promise<void> {
  await request(
    "/api/v1/admin/auth/logout",
    { method: "POST", token: accessToken },
    (value) => z.object({ success: z.boolean() }).parse(value),
  );
}

function toSession(value: z.infer<typeof sessionSchema>): AdminSessionSummary {
  return {
    displayName: value.display_name,
    capabilityLabel: value.capability_label,
    environmentKind: "demo",
    accessToken: value.access_token,
    refreshToken: value.refresh_token,
    accessExpiresAt: value.access_expires_at,
    refreshExpiresAt: value.refresh_expires_at,
  };
}

export function getInitialAdminSession(): Pick<
  AdminSessionSummary,
  "displayName" | "capabilityLabel" | "environmentKind"
> {
  return {
    displayName: FIXED_ADMIN_LOGIN,
    capabilityLabel: FIXED_ADMIN_CAPABILITY,
    environmentKind: "demo",
  };
}

File: backend/app/api/admin_auth.py
"""Fixed-account admin authentication and server-derived capability projection."""

from typing import Annotated

from fastapi import APIRouter, Header, Request

from app.api.dependencies import admin_subject, bearer_token, request_id
from app.schemas.auth import AdminLoginRequest, AdminMeData, AdminSessionData, RefreshRequest
from app.schemas.envelope import ApiEnvelope
from app.schemas.errors import ApiException

router = APIRouter(prefix="/api/v1/admin", tags=["admin-auth"])


@router.post("/auth/login")
async def login(request: Request, body: AdminLoginRequest) -> ApiEnvelope[AdminSessionData]:
    data = request.app.state.auth_service.login_admin(body.login_name, body.password)
    return ApiEnvelope.success(request_id(request), data=data)


@router.post("/auth/refresh")
async def refresh(request: Request, body: RefreshRequest) -> ApiEnvelope[object]:
    data = request.app.state.auth_service.refresh(
        body.refresh_token,
        expected_subject_type="admin",
    )
    if not isinstance(data, AdminSessionData):
        raise ApiException(401, "AUTH_REQUIRED")
    return ApiEnvelope.success(request_id(request), data=data)


@router.post("/auth/logout")
async def logout(
    request: Request,
    authorization: Annotated[str | None, Header()] = None,
) -> ApiEnvelope[dict[str, bool]]:
    token = bearer_token(authorization)
    request.app.state.auth_service.logout(token, expected_subject_type="admin")
    return ApiEnvelope.success(request_id(request), data={"success": True})


@router.get("/me")
async def me(
    request: Request,
    authorization: Annotated[str | None, Header()] = None,
) -> ApiEnvelope[AdminMeData]:
    subject = admin_subject(request, authorization)
    settings = request.app.state.settings
    data = AdminMeData(
        display_name="心理健康中心工作人员",
        capability_label="超级管理员",
        session_expires_at=subject.session_expires_at,
        environment_kind=settings.environment_kind.value,
    )
    return ApiEnvelope.success(request_id(request), data=data)
